### PR TITLE
D8CORE-3380 Allow media library widget to be sortable

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Use https://github.com/clauderic/react-sortable-hoc to provide sorting of the media library items.

# Need Review By (Date)
- 3/3

# Urgency
- Medium

# Steps to Test
1. checkout this branch
2. clear browser cache
3. create a basic page with a gallery paragraph type
4. add multiple images
5. verify you can sort and save the order of the items.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: js/src/Components/Atoms/FormDialog.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const FormDialog = ({open, title, formContent, onClose, ...props}) => {
         color: '#fff',
         position: 'sticky',
         top: 0,
-        zIndex: 2
+        zIndex: 1000
```
Comment (sherakama) [inline js/src/Components/Atoms/FormDialog.js:50]: That is very high. Does it need to be that high?

File: js/src/Components/Widgets/Media/MediaList.js [hunk 2]
```diff
@@ -4,6 +4,9 @@ import {FlexDiv} from "../../Atoms/FlexDiv";
 import {XButton} from "../../Atoms/XButton";
 import {Loader} from "../../Atoms/Loader";
 import {ErrorBoundary} from "../../Atoms/ErrorBoundary";
+import {SortableContainer, SortableElement} from "react-sortable-hoc";
```
Comment (sherakama) [inline js/src/Components/Widgets/Media/MediaList.js:7]: Looks like a neat package.

**Comment by imonroe** [comment]
Testing in my local, I am able to successfully create the gallery and view it in the page (and it looks nice), but when I go back to edit the gallery, I get a 500 error, and then I get a syntax error, both in the `WidgetManager.js`.  The editor for the gallery paragraph never loads.  This was using an incognito window to make sure browser cache was not an issue, and clearing the Drupal caches didn't affect it.

image

File: js/src/Components/Widgets/Media/MediaList.js [hunk 3]
```diff
@@ -21,24 +26,47 @@ const PreviewImageTag = styled.img`
...
+  );
+});
+
 /**
  * List of media items for media widget.
  */
 export const MediaList = ({selectedItems, onRemove, updateOrder}) => {
 
+  const onSortEnd = ({oldIndex, newIndex}) => {
+    updateOrder(arrayMove(selectedItems, oldIndex, newIndex));
```
Comment (sherakama) [inline js/src/Components/Widgets/Media/MediaList.js:59]: Neat package.

**Comment by sherakama** [comment]
Thanks @imonroe for testing.

The added packages and code looks fine overall from my perspective. Please continue to work through the implementation bugs.

File: js/src/Components/Atoms/FormDialog.js (hunk 1 above)
Comment (pookmish) [reply js/src/Components/Atoms/FormDialog.js:50]: it needs to be 1 below the media preview container and i tried that several hundred and still didn't work. so i just went there :man_shrugging:

File: js/src/Components/Widgets/Media/MediaList.js (hunk 3 above)
Comment (pookmish) [reply js/src/Components/Widgets/Media/MediaList.js:59]: yeah i liked that one.

**Review by imonroe — APPROVED** [review]
After working with Mike for a while on this, I was still seeing a bug with deletions of images in an existing gallery, which I was able to duplicate on Browserstack with a Tugboat test server and both FF and Chrome on Windows and Mac Big Sur.  Mike was unable to duplicate with the same setup, however.

We'll merge this for now and give it a thorough testing on the cardinalsites test environment tomorrow, with more people trying it out to make sure it's not a problem.

**Pull request merged**